portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,35 +14,99 @@
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    String convertTojson(ArrayList<String> comments) {
-        Gson gson = new Gson();
-        String convertedTojson = gson.toJson(comments);
-        return convertedTojson;
-    }
+  //this is the number of comments that should
+  //be printed when the comments page is loaded
+  private int commentCount = 99;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> messages = new ArrayList<>();
 
-    messages.add(0, "Star Wars is awesome!");
-    messages.add(1, "I will become the next UN secretary general.");
-    messages.add(2, "I am to go back to school.");
+    Query query = new Query("Comments").addSort("time", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    //The loop below is storing Comment objects to make it
+    //easier for the function displayComments to create a list element
+    //on HTML.
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(commentCount))) {
+      long id = entity.getKey().getId();
+      String Statement = (String) entity.getProperty("comment");
+      long Time = (long) entity.getProperty("time");
+
+      Comment messages = new Comment(id, Statement, Time);
+      comments.add(messages);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      long time = System.currentTimeMillis();
+      String statement = comment(request, "texts");
+      commentCount = maxComments(request, "preferedNumber");
+
+      if (commentCount < 0) {
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter an integer between 0 and 100.");
+        return;
+      }
+
+      addcomment(statement, time);
+
+      response.sendRedirect("/comments.html");
+  }
+
+  private String comment(HttpServletRequest request, String message) {
+    return request.getParameter(message);
+  }
 
-    String comments = messages.get((int) (Math.random() * messages.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(comments);
+  private void addcomment(String statements, long timestamp) {
```

Comment:
I would expect this to be 'addComment' in java style.

Reply:
Thanks. I didn't realize it